poetry_dynamic_versioning/plugin.py

```diff
@@ -81,8 +81,9 @@ def _apply_version_via_plugin(
     io: bool = True
     # fmt: on
 ) -> None:
+    project_section = poetry.pyproject.data.get("project", {})
     name = _get_and_apply_version(
-        name=poetry.local_config["name"],
+        name=project_section.get("name") or poetry.local_config("name"),
```

Comment:
The change from `local_config["name"]` to `local_config("name")` is causing the tests to fail:

```
FAILED tests/test_integration.py::test_plugin_enabled - RuntimeError: The command 'poetry build' returned code 1. Output:
'dict' object is not callable
```

Also, I think it's probably better if `_get_and_apply_version` derives the name and original version from the `pyproject` argument, to centralize that logic. That way, `_get_and_apply_version` can be aware of which `name` was set and pass that information to `_apply_version` so it can determine which `version` field to set.